services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -196,25 +196,40 @@ async def sync_registry_task(app: FastAPI) -> None:
             # (templates are published to GUESTs, so their services must be also accessible)
             await _ensure_published_templates_accessible(engine, default_product)
 
-            await asyncio.sleep(app.state.settings.background_task_rest_time)
+            await asyncio.sleep(app.state.settings.CATALOG_BACKGROUND_TASK_REST_TIME)
 
-        except CancelledError:
+        except asyncio.CancelledError:
             # task is stopped
-            logger.debug("Catalog background task cancelled", exc_info=True)
-            return
+            logger.info("registry syncing task cancelled", exc_info=True)
+            raise
 
         except Exception:  # pylint: disable=broad-except
-            logger.exception("Error while processing services entry")
+            if not app.state.registry_syncer_running:
+                logger.warning("registry syncing task forced to stop")
+                break
+            logger.exception(
+                "Unexpected error while syncing registry entries, restarting now..."
+            )
             # wait a bit before retrying, so it does not block everything until the director is up
-            await asyncio.sleep(app.state.settings.background_task_wait_after_failure)
+            await asyncio.sleep(app.state.settings.BACKGROUND_TASK_WAIT_AFTER_FAILURE)
 
 
 async def start_registry_sync_task(app: FastAPI) -> None:
-    task = asyncio.ensure_future(sync_registry_task(app))
+    # FIXME: added this variable to overcome the state in which the
+    # task cancelation is ignored and the exceptions enter in a loop
+    # that never stops the background task. This flag is an additional
+    # mechanism to enforce stopping the background task
+    app.state.registry_syncer_running = True
+    task = asyncio.create_task(sync_registry_task(app))
     app.state.registry_sync_task = task
+    logger.info("registry syncing task started")
 
 
 async def stop_registry_sync_task(app: FastAPI) -> None:
     task = app.state.registry_sync_task
-    task.cancel()
-    await task
+    with suppress(asyncio.CancelledError):
+        app.state.registry_syncer_running = False
+        task.cancel()
```

Comment:
is this done to break the 60 second sleep period?

Reply:
no, there is currently an issue sometimes in these background tasks that do not properly cancel (see for instance services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py